lib/modules/manager/pep621/processors/uv.ts

```diff
@@ -191,3 +198,34 @@ function generateCMD(updatedDeps: Upgrade[]): string {
 
   return `${uvUpdateCMD} ${deps.map((dep) => `--upgrade-package ${quote(dep)}`).join(' ')}`;
 }
+
+function getMatchingHostRule(url: string | undefined): HostRule {
+  const scopedMatch = find({ hostType: PypiDatasource.id, url });
+  return is.nonEmptyObject(scopedMatch) ? scopedMatch : find({ url });
+}
+
+function getUvExtraIndexUrl(updatedDeps: Upgrade[]): NodeJS.ProcessEnv {
```

Comment:
It might be possible that there are index URLs _not_ in one of the Upgrades which uv needs, but we can wait for that edge case and add/fix it later

Reply:
Rrrright, the idea I had in my head in the beginning was to just pry out all of the `registryUrl` things from `packageRules` for `pypi`, but not knowing the code very well, I didn't figure out how to do that 😅